### PR TITLE
[KHWEBA-95]Edit tag flow

### PR DESCRIPTION
**Pull request by OlgaOrlova**

### CODE REVIEW

File: src/app/tags/tag-dialog/tag-dialog.component.ts [hunk 1]
```diff
@@ -28,16 +28,33 @@ export class TagDialogComponent implements OnInit {
...
   }
 
   submit(form: any) {
-    this.matDialogRef.close(form.value);
+    const val = form.value;
+    let id = "";
+    if (this.data.dataToEdit) {
+      id = this.data.dataToEdit.id;
+    } else {
+      id = form.id;
```
Comment (gift-a) [inline src/app/tags/tag-dialog/tag-dialog.component.ts]: what does form.id equal?

File: src/app/tags/tags/tags.component.ts [hunk 2]
```diff
@@ -71,7 +74,15 @@ export class TagsComponent implements OnInit {
       });
```
Comment (gift-a) [inline src/app/tags/tags/tags.component.ts:76]: - `if (transactions.length > 0)` the same as `if (transactions.length)`
- you will try to delete transactions/tag on each transactions list change. try to use operator `first` for observable

File: src/app/tags/tags/tags.component.ts [hunk 3]
```diff
@@ -71,7 +74,15 @@ export class TagsComponent implements OnInit {
...
+  editTag(tag) {
+    const editTagDialog = this.dialog.open(TagDialogComponent, {
+      data: { action: "Edit", dataToEdit: tag, tags: this.tags },
+      minWidth: "50%"
+    });
+    editTagDialog.afterClosed().subscribe(res => {
+      if (res) {
+        this.tagsService.updateData(res).subscribe();
+      }
+    });
```
Comment (gift-a) [inline src/app/tags/tags/tags.component.ts:88]: rows 78 - 86 are almost duplicate. isn't is better to make one method for opening dialog with variables `action`, `tags` and `dataToEdit`?

File: src/app/tags/tags/tags.component.ts (hunk 2 above)
Comment (OlgaOrlova) [reply src/app/tags/tags/tags.component.ts:76]: 1. It makes code more readable
2. I do unsubscribe after delete operation completes

File: src/app/tags/tags/tags.component.ts (hunk 3 above)
Comment (OlgaOrlova) [reply src/app/tags/tags/tags.component.ts:88]: I don't think, that pass "action" in dialog and check this variable in result is better than two separate methods.

File: src/app/tags/tags/tags.component.ts (hunk 2 above)
Comment (gift-a) [reply src/app/tags/tags/tags.component.ts:76]: 2. Try to log your deletion. You will see much more actions then you expect

File: src/app/tags/tags/tags.component.ts (hunk 3 above)
Comment (gift-a) [reply src/app/tags/tags/tags.component.ts:88]: Ok

File: src/app/tags/tag-dialog/tag-dialog.component.ts [hunk 4]
```diff
@@ -28,16 +28,33 @@ export class TagDialogComponent implements OnInit {
...
   }
 
   submit(form: any) {
-    this.matDialogRef.close(form.value);
+    const val = form.value;
+    let id = "";
+    if (this.data.dataToEdit) {
+      id = this.data.dataToEdit.id;
+    } else {
+      id = val.id;
```
Comment (gift-a) [inline src/app/tags/tag-dialog/tag-dialog.component.ts]: what does val.id equal?

**Review by gift-a — APPROVED** [review]

**Pull request merged**